### PR TITLE
Add testing for various params that pass through Contact_BAO_Query::optionValueQuery

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Adds testing & a little cleanup (includes #15995)

Before
----------------------------------------
Special handling for gender_id etc in multiple places

After
----------------------------------------
Just in one place - the reversal of the false 'type' variable in the metadata

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton needs a rebase

**Comment by eileenmcnaughton** [comment]
@seamuslee001 ok - done

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -5907,8 +5905,8 @@ public function optionValueQuery(
         // Special handling for on_hold, so that we actually use the 'where'
         // property in order to limit the query by the on_hold status of the email,
         // instead of using email.id which would be nonsensical.
-        if ($field['name'] == 'on_hold') {
-          $wc = "{$field['where']}";
+        if ($field['name'] === 'on_hold') {
+          $wc = $field['where'];
```
Comment (seamuslee001) [inline CRM/Contact/BAO/Query.php]: @eileenmcnaughton any reason for the removal of the quotes here?

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/Query.php]: @seamuslee001 readability

**Comment by seamuslee001** [comment]
(*CiviCRM Review Template WORD-1.2*)



* General standards
    * ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) Pass
    * ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) Pass
    * ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) Pass
    * ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) Pass did some r-run on advanced search looks fine, backed up sensibly with unit tests
* Developer standards
    * ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) Pass
    * ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) Pass
    * ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) Pass
    * ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) Unknown

**Pull request merged**